internal/expr/prepare.go

```diff
@@ -77,14 +80,63 @@ func prepareInput(ti typeNameToInfo, p *inputPart) (tm typeMember, err error) {
 			return nil, fmt.Errorf(`type %q not passed as a parameter (have "%s")`, p.sourceType.prefix, strings.Join(ts, `", "`))
 		}
 	}
-
+	if _, ok = info.(*sliceInfo); ok {
+		return nil, fmt.Errorf(`cannot use slice type %q outside of IN clause`, info.typ().Name())
```

Comment:
This seems rather limiting. Per above note, I think we should first talk about what is the overall representation and model of slices, and then see if that model works well with IN and other cases where there are lists involved on the SQL side.

Reply:
See update comment below.